src/ORM/DataList.php

```diff
@@ -1030,28 +1030,37 @@ private function getEagerLoadVariables(string $eagerLoadRelation): array
         $hasManyIDField = null;
         $manyManyLastComponent = null;
         for ($i = 0; $i < count($relations) - 1; $i++) {
-            $hasOneComponent = $schema->hasOneComponent($dataClasses[$i], $relations[$i + 1]);
+            $parentDataClass = $dataClasses[$i];
+            $relationName = $relations[$i + 1];
+            $hasOneComponent = $schema->hasOneComponent($parentDataClass, $relationName);
             if ($hasOneComponent) {
                 $dataClasses[] = $hasOneComponent;
                 $hasOneIDField = $relations[$i + 1] . 'ID';
                 continue;
             }
-            $belongsToComponent = $schema->belongsToComponent($dataClasses[$i], $relations[$i + 1]);
+            $belongsToComponent = $schema->belongsToComponent($parentDataClass, $relationName);
             if ($belongsToComponent) {
                 $dataClasses[] = $belongsToComponent;
-                $belongsToIDField = $schema->getRemoteJoinField($dataClasses[$i], $relations[$i + 1], 'belongs_to');
+                $belongsToIDField = $schema->getRemoteJoinField($parentDataClass, $relationName, 'belongs_to');
                 continue;
             }
-            $hasManyComponent = $schema->hasManyComponent($dataClasses[$i], $relations[$i + 1]);
+            $hasManyComponent = $schema->hasManyComponent($parentDataClass, $relationName);
             if ($hasManyComponent) {
                 $dataClasses[] = $hasManyComponent;
-                $hasManyIDField = $schema->getRemoteJoinField($dataClasses[$i], $relations[$i + 1], 'has_many');
+                $hasManyIDField = $schema->getRemoteJoinField($parentDataClass, $relationName, 'has_many');
                 continue;
             }
             // this works for both many_many and belongs_many_many
-            $manyManyComponent = $schema->manyManyComponent($dataClasses[$i], $relations[$i + 1]);
+            $manyManyComponent = $schema->manyManyComponent($parentDataClass, $relationName);
             if ($manyManyComponent) {
                 $dataClasses[] = $manyManyComponent['childClass'];
+                $manyManyComponent['extraFields'] = $schema->manyManyExtraFieldsForComponent($parentDataClass, $relationName) ?: [];
+                if (is_a($manyManyComponent['relationClass'], ManyManyThroughList::class, true)) {
+                    $manyManyComponent['joinClass'] = $manyManyComponent['join'];
+                    $manyManyComponent['join'] = $schema->baseDataTable($manyManyComponent['joinClass']);
+                } else {
+                    $manyManyComponent['joinClass'] = null;
+                }
```

Comment:
Make `join` always be the join _table_, and add the join _class_ separately.
Also fetch the extra fields schema up front.